api.yml

```diff
@@ -19,9 +19,97 @@ paths:
     post:
       tags:
         - internal
-      summary: Verifies a credential and returns its verification status.
+      summary: Verify a given credential.
```

Comment:
Can we verify presentations, too? If yes, do we expect to call the API multiple times if the presentation contains multiple credentials inside it? Or do we call it once, and all credentials inside the presentation are checked? If we call multiple times, what logic do we use (if any) to verify coherence (that is, each credential hangs together with the others in an overall gestalt)?

Reply:
+1 to verify presentations too, see https://github.com/w3c-ccg/vc-verifier-http-api/issues/5

Reply:
> Can we verify presentations, too?

Yes, that's the intent, eventually. During the Issuer HTTP API discussion, folks seemed to want to focus on credentials first, and then fold presentations in once we got agreement on credentials. I was following the same playbook here. I tried to introduce presentations in my PR on the Issuer HTTP API and that led to push back, so I'm trying to start simpler here.

My expectation is that in another PR (if this one makes it), we'll add "verifiablePresentation" as a potential input to this endpoint. There might be alternate arguments from @OR13 to do .../verifications/credentials and .../verifications/presentations ... which decomposes the functionality in a way that @dhh1128 was suggesting (microservices approach) somewhere else on the Internet. I see benefits of doing it that way as well.

> If yes, do we expect to call the API multiple times if the presentation contains multiple credentials inside it? Or do we call it once, and all credentials inside the presentation are checked?

I'd like us to have an option where we call it once. Keep in mind that if we are also providing an endpoint to do credentials, we could do the credentials one by one... but that seems like a pain to put the developer through. It's certainly something we need to discuss in the future.

The downside, of course, with providing a single shot verification endpoint is the potential DoS vector (e.g., due to having to go out to a ledger to retrieve keys that are not cached). There are mitigations to that (the verification reference stuff that this API does)... but then that complicates the API. In any case, we'll worry about that in another PR.

Reply:
I'd prefer not to do presentations until have have credentials working... 1 thing at a time.

Reply:
> I'd like us to have an option where we call it once. 
I agree; it is unclear what return status the API would return in the case where some credentials pass and others fail (200 or 400?).

One option is to return status 200 if the input was valid and the process able to run, but the return content would specify there was an error in validating the signature of a particular credential.

Other option is we clarify 400 is returned in case of *any* of the credentials failing (arguable signature failing == bad input) but the error message may contain "some credentials passed" results...

Or maybe the final option is the API validates multiple credentials as a package; therefore if one is bad the package is bad (400) - if the caller wants individual results, the API should be called multiple times separately.

(I'm also Ok with opening an issue to track this subtly if there is no easy clarification).

Reply:
This is being tracked in issue #5.